In [7]:

import numpy as np
from scipy.optimize import linprog 

A = np.array([
    [20, 15, 25],  
    [10, 12, 15], 
    [5, 8, 10]    
])


B = np.array([1000, 800, 600])


C = np.array([-1, -1, -1])


result = linprog(C, A_ub=A, b_ub=B)

optimal_units = result.x
optimal_total_units = -result.fun 

print("Optimal units for A, B, C:", optimal_units)
print("Optimal total units:", optimal_total_units)




Optimal units for A, B, C: [-2.27373675e-14  6.66666667e+01  0.00000000e+00]
Optimal total units: 66.66666666666667


In [8]:
import numpy as np


matrix_A = np.array([[50, 20], [30, 40]])


matrix_B = np.array([[45, 10], [25, 35]])


det_A = np.linalg.det(matrix_A)

det_B = np.linalg.det(matrix_B)

print("Determinant of matrix A:", det_A)
print("Determinant of matrix B:", det_B)


Determinant of matrix A: 1399.9999999999998
Determinant of matrix B: 1325.0


b. 
Determinan dari matriks dapat memberikan informasi tentang kualitas kinerja perusahaan, terutama dalam konteks ini di mana matriks mewakili kinerja penjualan dan produksi bulanan. Detilnya, det(D) dapat memberikan petunjuk tentang seberapa baik suatu perusahaan telah berkinerja.

Determinan Matriks A:

(
�
)
=
(
50
×
40
)
−
(
20
×
30
)
=
2000
−
600
=
1400
det(A)=(50×40)−(20×30)=2000−600=1400

Determinan Matriks B:

(
�
)
=
(
45
×
35
)
−
(
10
×
25
)
=
1575
−
250
=
1325
det(B)=(45×35)−(10×25)=1575−250=1325

Analisis:

Determinan Positif (det > 0): Ini menunjukkan bahwa perusahaan A dan B keduanya memiliki kinerja yang baik. Determinan yang positif biasanya dikaitkan dengan kualitas kinerja yang baik.

Perbandingan Determinan: Det(A) > Det(B) menunjukkan bahwa, berdasarkan matriks ini, perusahaan A memiliki kinerja yang lebih baik daripada perusahaan B. Hal ini dapat diartikan bahwa perusahaan A mungkin lebih efisien dalam penjualan dan produksi bulanannya.

Korelasi dengan Kualitas Kinerja: Meskipun determinan memberikan indikasi umum tentang kualitas kinerja, tetapi tidak memberikan rincian tentang aspek-aspek spesifik dari penjualan dan produksi. Jadi, sementara perusahaan A mungkin memiliki kinerja keseluruhan yang lebih baik, aspek spesifik dari kinerja harus dianalisis lebih lanjut untuk mendapatkan pemahaman yang lebih mendalam.

In [9]:
import numpy as np


A = np.array([[50, 20], [30, 40]])
B = np.array([[45, 10], [25, 35]])


try:
    A_inv = np.linalg.inv(A)
    print("Inverse of matrix A:")
    print(A_inv)
except np.linalg.LinAlgError:
    print("Matrix A is singular, and its inverse does not exist.")


try:
    B_inv = np.linalg.inv(B)
    print("\nInverse of matrix B:")
    print(B_inv)
except np.linalg.LinAlgError:
    print("Matrix B is singular, and its inverse does not exist.")


Inverse of matrix A:
[[ 0.02857143 -0.01428571]
 [-0.02142857  0.03571429]]

Inverse of matrix B:
[[ 0.02641509 -0.00754717]
 [-0.01886792  0.03396226]]


d. Untuk memahami implikasi dari hasil invers matriks A dan B pada operasi komputasi, mari kita temukan invers dari matriks tersebut.

Misalkan matriks A:
�
=
[
50
20
30
40
]
A=[ 
50
30
​
  
20
40
​
 ]

Dan matriks B:
�
=
[
45
10
25
35
]
B=[ 
45
25
​
  
10
35
​
 ]

Invers dari matriks 2x2 
[
�
�
�
�
]
[ 
a
c
​
  
b
d
​
 ] diberikan oleh rumus:
Invers
=
1
�
�
−
�
�
[
�
−
�
−
�
�
]
Invers= 
ad−bc
1
​
 [ 
d
−c
​
  
−b
a
​
 ]

Sekarang, mari temukan invers matriks A dan B:

Untuk matriks A:
Invers A
=
1
(
50
×
40
)
−
(
20
×
30
)
[
40
−
20
−
30
50
]
Invers A= 
(50×40)−(20×30)
1
​
 [ 
40
−30
​
  
−20
50
​
 ]

Dan untuk matriks B:
Invers B
=
1
(
45
×
35
)
−
(
10
×
25
)
[
35
−
10
−
25
45
]
Invers B= 
(45×35)−(10×25)
1
​
 [ 
35
−25
​
  
−10
45
​
 ]

Sekarang, mari bahas implikasi dari matriks invers ini pada operasi komputasi:

Perkalian Matriks dengan Invers:

Untuk matriks A, hasil perkalian A dengan inversnya akan menghasilkan matriks identitas.
�
×
Invers A
=
Matriks Identitas
A×Invers A=Matriks Identitas
Hal yang sama berlaku untuk matriks B:
�
×
Invers B
=
Matriks Identitas
B×Invers B=Matriks Identitas
Properti ini berguna dalam menyelesaikan sistem persamaan linear dan dapat menyederhanakan beberapa komputasi.

Penyelesaian Sistem Persamaan Linear:

Dengan sistem persamaan linear yang direpresentasikan oleh matriks A dan B, Anda dapat menggunakan inversnya untuk menemukan solusinya.
�
=
Invers A
×
B
X=Invers A×B
Ini didasarkan pada fakta bahwa 
�
×
�
=
�
A×X=B menyiratkan 
�
=
Invers A
×
B
X=Invers A×B.

Efisiensi Komputasi:

Menggunakan matriks invers dapat lebih efisien secara komputasional daripada menyelesaikan langsung sistem persamaan, terutama ketika berurusan dengan matriks besar.